lightning/restore/restore.go

```diff
@@ -299,51 +298,37 @@ outside:
 
 func (rc *RestoreController) restoreSchema(ctx context.Context) error {
 	if !rc.cfg.Mydumper.NoSchema {
+		// CONFUSED: how we control concurrency via configuration?
+		concurrency := 16
```

Comment:
the configuration file is read into the `Config` struct in `lightning/config/config.go` via reflection. so you just need to add a field in the appropriate place.

but let's first verify if adding concurrency here has any improvement before considering generalization.

Reply:
@kennytm It makes sense, we need run some benchmark scripts?

Reply:
yes. basically create like 300 small tables and compare the time needed before and after this PR.

Reply:
@kennytm With any exist tool(e.g 300 small tables generator)?

Reply:
you can refer to this snippet in BR: https://github.com/pingcap/br/blob/cf1c9b245bf550ba0b900619a143856186090db3/tests/br_300_small_tables/run.sh#L20-L28

Reply:
@kennytm PR updated, PTAL. time costs report as below:

### Before

```

________________________________________________________
Executed in  211.51 secs   fish           external 
   usr time   76.28 secs  187.00 micros   76.28 secs 
   sys time   44.62 secs  617.00 micros   44.62 secs 

[2020/12/08 17:06:33.389 +08:00] [INFO] [restore.go:964] ["restore all tables data completed"] [takeTime=1m9.093660687s] []
[2020/12/08 17:06:33.389 +08:00] [INFO] [restore.go:745] ["everything imported, stopping periodic actions"]
[2020/12/08 17:06:33.389 +08:00] [INFO] [restore.go:1409] ["skip full compaction"]
[2020/12/08 17:06:33.411 +08:00] [INFO] [restore.go:294] ["the whole procedure completed"] [takeTime=1m45.325956477s] []
[2020/12/08 17:06:33.411 +08:00] [INFO] [main.go:95] ["tidb lightning exit"]
[2020/12/08 17:06:33.411 +08:00] [INFO] [checksum.go:425] ["service safe point keeper exited"]
```

### After

```

________________________________________________________
Executed in  213.24 secs   fish           external 
   usr time   78.08 secs  140.00 micros   78.08 secs 
   sys time   44.92 secs  475.00 micros   44.92 secs 

[2020/12/08 16:55:15.821 +08:00] [INFO] [restore.go:820] ["restore all tables data completed"] [takeTime=1m9.754043571s] []
[2020/12/08 16:55:15.821 +08:00] [INFO] [restore.go:601] ["everything imported, stopping periodic actions"]
[2020/12/08 16:55:15.821 +08:00] [INFO] [restore.go:1265] ["skip full compaction"]
[2020/12/08 16:55:15.840 +08:00] [INFO] [restore.go:293] ["the whole procedure completed"] [takeTime=1m42.140242288s] []
[2020/12/08 16:55:15.840 +08:00] [INFO] [main.go:95] ["tidb lightning exit"]
[2020/12/08 16:55:15.840 +08:00] [INFO] [checksum.go:425] ["service safe point keeper exited"]
```

Orz hmmmm.... it is not a good result.

Reply:
please add these benchmark results to the pr description before the `Check List` section.

Reply:
Ok.